File: projects/fizzbuzz/python/b-cornett/fizzBuzz.py
#FizzBuzz program, prints Fizz if the mod operator returns 0 for 3, Buzz if the mod operator returns 0, and FizzBuzz
#if both of them return 0

#=================================
#Function for FizzBuzz iteration
#=================================
def fizzbuzz(number):
    output = ""
    divis = False
    if (number % 3) == 0:
        output += "Fizz"
        divis = True
    if (number % 5) == 0:
        output += "Buzz"
        divis = True
    if (number % 7) == 0:
        output += "Bazz"
        divis = True
    if divis != True:
        return (number)
    else:
        return (output)

#=================================
#Driver section of python script
#=================================
def main():
        try:
            inpNumber = int(input("Enter number for FizzBuzz function: "))
            if inpNumber <= 0:
                print("INVALID INPUT: Please enter a positive integer.")
            else:
                counter = 1
                # While loop for iterating through user input
                while counter <= inpNumber:
                    print(fizzbuzz(counter))
                    counter = counter + 1
        except ValueError:
            print("INVALID INPUT: Please enter a positive integer.")

#=============================
# Execution of the program
#=============================

if __name__ == '__main__':
    main()

File: projects/fizzbuzz/fortran/seeley/fizzbuzz.f90
program fizzbuzz

   implicit none

   integer :: n               !maximum integer number from user input
   integer :: i = 0           !loop iterator
   character*8 :: stringout   !output string on each iteration

   !get integer up to which FizzBuzz will go
   print *, "Please enter a number up to which FizzBuzz shall be played:  "
   read *, n
   print *, "You chose:  ", n

   !print each number appropriately
   do i=1, n
      stringout = ""
      if (mod(i, 3) == 0) then
         stringout = trim(stringout) // "Fizz"
      endif
      if (mod(i, 5) == 0) then
         stringout = trim(stringout) // "Buzz"
      endif
      if (stringout == "") then
         write (stringout, "(1i4)") i
      endif

      print *, stringout
   
   end do
   

end program fizzbuzz
